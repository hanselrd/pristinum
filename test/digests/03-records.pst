struct Point:
    x @f64;
    y @f64;
end

func main() @void:
    p @Point := Point;
    p x .`` = 1.0;
    p y .`` = 2.0;
end