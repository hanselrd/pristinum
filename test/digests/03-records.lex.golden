[ PT
    ( Pn 0 1 1 )
    ( TK
        ( TokSymbol
            { tsText = "struct"
            , tsID = 48
            }
        )
    )
, PT
    ( Pn 7 1 8 )
    ( T_IDENT "Point" )
, PT
    ( Pn 12 1 13 )
    ( TK
        ( TokSymbol
            { tsText = ":"
            , tsID = 18
            }
        )
    )
, PT
    ( Pn 18 2 5 )
    ( T_IDENT "x" )
, PT
    ( Pn 20 2 7 )
    ( TK
        ( TokSymbol
            { tsText = "@"
            , tsID = 29
            }
        )
    )
, PT
    ( Pn 21 2 8 )
    ( TK
        ( TokSymbol
            { tsText = "f64"
            , tsID = 40
            }
        )
    )
, PT
    ( Pn 24 2 11 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 30 3 5 )
    ( T_IDENT "y" )
, PT
    ( Pn 32 3 7 )
    ( TK
        ( TokSymbol
            { tsText = "@"
            , tsID = 29
            }
        )
    )
, PT
    ( Pn 33 3 8 )
    ( TK
        ( TokSymbol
            { tsText = "f64"
            , tsID = 40
            }
        )
    )
, PT
    ( Pn 36 3 11 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 38 4 1 )
    ( TK
        ( TokSymbol
            { tsText = "end"
            , tsID = 39
            }
        )
    )
, PT
    ( Pn 43 6 1 )
    ( TK
        ( TokSymbol
            { tsText = "func"
            , tsID = 42
            }
        )
    )
, PT
    ( Pn 48 6 6 )
    ( T_IDENT "main" )
, PT
    ( Pn 52 6 10 )
    ( TK
        ( TokSymbol
            { tsText = "("
            , tsID = 6
            }
        )
    )
, PT
    ( Pn 53 6 11 )
    ( TK
        ( TokSymbol
            { tsText = ")"
            , tsID = 7
            }
        )
    )
, PT
    ( Pn 55 6 13 )
    ( TK
        ( TokSymbol
            { tsText = "@"
            , tsID = 29
            }
        )
    )
, PT
    ( Pn 56 6 14 )
    ( TK
        ( TokSymbol
            { tsText = "void"
            , tsID = 51
            }
        )
    )
, PT
    ( Pn 60 6 18 )
    ( TK
        ( TokSymbol
            { tsText = ":"
            , tsID = 18
            }
        )
    )
, PT
    ( Pn 66 7 5 )
    ( T_IDENT "p" )
, PT
    ( Pn 68 7 7 )
    ( TK
        ( TokSymbol
            { tsText = "@"
            , tsID = 29
            }
        )
    )
, PT
    ( Pn 69 7 8 )
    ( T_IDENT "Point" )
, PT
    ( Pn 75 7 14 )
    ( TK
        ( TokSymbol
            { tsText = ":="
            , tsID = 19
            }
        )
    )
, PT
    ( Pn 78 7 17 )
    ( T_IDENT "Point" )
, PT
    ( Pn 83 7 22 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 89 8 5 )
    ( T_IDENT "p" )
, PT
    ( Pn 90 8 6 )
    ( TK
        ( TokSymbol
            { tsText = "."
            , tsID = 16
            }
        )
    )
, PT
    ( Pn 91 8 7 )
    ( T_IDENT "x" )
, PT
    ( Pn 92 8 8 )
    ( TK
        ( TokSymbol
            { tsText = "="
            , tsID = 24
            }
        )
    )
, PT
    ( Pn 94 8 10 )
    ( TD "1.0" )
, PT
    ( Pn 97 8 13 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 103 9 5 )
    ( T_IDENT "p" )
, PT
    ( Pn 104 9 6 )
    ( TK
        ( TokSymbol
            { tsText = "."
            , tsID = 16
            }
        )
    )
, PT
    ( Pn 105 9 7 )
    ( T_IDENT "y" )
, PT
    ( Pn 107 9 9 )
    ( TK
        ( TokSymbol
            { tsText = "="
            , tsID = 24
            }
        )
    )
, PT
    ( Pn 109 9 11 )
    ( TD "2.0" )
, PT
    ( Pn 112 9 14 )
    ( TK
        ( TokSymbol
            { tsText = ";"
            , tsID = 20
            }
        )
    )
, PT
    ( Pn 114 10 1 )
    ( TK
        ( TokSymbol
            { tsText = "end"
            , tsID = 39
            }
        )
    )
]